[Bug]: `r/aws_amplify_app`: `environment_variables` cannot be unset once configured

### Terraform Core Version

1.9.5

### AWS Provider Version

5.67.0

### Affected Resource(s)

- `aws_amplify_app`

### Expected Behavior

When the `environment_variables` argument is removed, the existing values should be removed on the remote resource.

### Actual Behavior

Terraform fails with an error like:

```
        Error: updating Amplify App (d3uac3jivrfkjf): operation error Amplify: UpdateApp, https response error StatusCode: 400, RequestID: e0eb1198-08a2-4d93-8b23-bbcc2201eb1f, BadRequ
estException: Environment variables cannot have an empty key.
```

### Relevant Error/Panic Output Snippet

_No response_

### Terraform Configuration Files

```terraform
resource "aws_amplify_app" "test" {
  name = "jb-test"

  # remove this on a subsequent apply
  environment_variables = {
    ENVVAR1 = "1",
  }
}
```

### Steps to Reproduce

1. Apply the configuration above.
2. Remove the `environment_variables` argument
3. Observe error on subsequent apply.

### Debug Output

_No response_

### Panic Output

_No response_

### Important Factoids

After some initial investigation, it isn't apparent how to remove existing environment variables with the AWS SDK for Go V2 now that empty keys are no longer permitted (at one point this worked to remove values with AWS SDK for Go V1, but now this pattern does not work with either SDK). A bug report has been filed here: https://github.com/aws/aws-sdk-go-v2/issues/2788.

The following acceptance test captures this workflow:

```console
% make testacc PKG=amplify TESTS=TestAccAmplify_serial/App/EnvironmentVariables
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.6 test ./internal/service/amplify/... -v -count 1 -parallel 20 -run='TestAccAmplify_serial/App/EnvironmentVariables'  -timeout 360m
=== RUN   TestAccAmplify_serial
=== PAUSE TestAccAmplify_serial
=== CONT  TestAccAmplify_serial
=== RUN   TestAccAmplify_serial/App
=== RUN   TestAccAmplify_serial/App/EnvironmentVariables
    app_test.go:427: Step 4/4 error: Error running apply: exit status 1

        Error: updating Amplify App (d3uac3jivrfkjf): operation error Amplify: UpdateApp, https response error StatusCode: 400, RequestID: e0eb1198-08a2-4d93-8b23-bbcc2201eb1f, BadRequ
estException: Environment variables cannot have an empty key.

          with aws_amplify_app.test,
          on terraform_plugin_test.tf line 12, in resource "aws_amplify_app" "test":
          12: resource "aws_amplify_app" "test" {

--- FAIL: TestAccAmplify_serial (25.38s)
    --- FAIL: TestAccAmplify_serial/App (25.37s)
        --- FAIL: TestAccAmplify_serial/App/EnvironmentVariables (25.37s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/amplify    31.543s
```

### References

- Depends on https://github.com/aws/aws-sdk-go-v2/issues/2788

### Would you like to implement a fix?

None

--- a/.changelog/39397.txt
+++ b/.changelog/39397.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_amplify_app: Fix failure when unsetting the `environment_variables` argument
+```

--- a/internal/service/amplify/app.go
+++ b/internal/service/amplify/app.go
@@ -545,7 +545,10 @@ func resourceAppUpdate(ctx context.Context, d *schema.ResourceData, meta interfa
 			if v := d.Get("environment_variables").(map[string]interface{}); len(v) > 0 {
 				input.EnvironmentVariables = flex.ExpandStringValueMap(v)
 			} else {
-				input.EnvironmentVariables = map[string]string{"": ""}
+				// To remove environment variables, set the key to a single space
+				// character and the value to an empty string.
+				// Ref: https://github.com/aws/aws-sdk-go-v2/issues/2788
+				input.EnvironmentVariables = map[string]string{" ": ""}
 			}
 		}
 

